capabilities/ecr-webhook/lib/ecr.ts

```diff
@@ -78,17 +74,19 @@ export async function createRepos(
 
   const region = process.env.AWS_REGION;
 
-  // Create repositories for private ECR registry
-  if (privateECRURLPattern.test(registryURL)) {
+  if (region === undefined) {
+    throw new Error("AWS_REGION environment variable is not defined.");
+  }
+
+  if (isPrivateECRURL(registryURL)) {
     const accountId = getAccountId(registryURL);
     const ecrPrivate = new ECRPrivate(region);
 
     Log.info("Attempting to create ECR repositories");
     await ecrPrivate.createRepositories(repoNames, accountId);
   }
 
-  // Create repositories for public ECR registry
-  if (publicECRURLPattern.test(registryURL)) {
+  if (isPublicECRURL(registryURL)) {
     const ecrPublic = new ECRPublic(region);
 
     Log.info("Attempting to create ECR repositories");
```

Comment:
The very first thing that happens when the webhook is triggered is it checks if the zarf state secret has a valid ECR URL:

```typescript
When(a.Secret)
  .IsCreatedOrUpdated()
  .InNamespace("zarf")
  .WithLabel("package-deploy-info")
  .Mutate(async request => {
    const result = await isECRregistry();

    if (!result.isECR) {
      throw new Error(
        `A valid ECR URL was not found in the Zarf state secret: ${result.registryURL}\n
        Please provide a valid ECR registry URL.\n
        Example: '123456789012.dkr.ecr.us-east-1.amazonaws.com'`,
      );
    }

```

So the registry URL is validated long before we ever call this function to create repos. Do you think it would be better to move the validation logic inside of the function instead?

Reply:
I do think it's best to have the validation as early in the process as possible though

Reply:
added validation for the registry URL prior to calling `isPublic` or `isPrivate`